SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -555,38 +555,35 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
 	return AlwaysInterestedConstraint;
 }
 
-QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
+QueryConstraint CreateOrConstraint(const TArray<Worker_ComponentId> ComponentIds)
 {
-	QueryConstraint SnapshotEntitiesConstraint;
-
-	Worker_ComponentId ComponentIds[] = { SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID,
-										  SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID,
-										  SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID };
+	QueryConstraint OrConstraint;
 
 	for (Worker_ComponentId ComponentId : ComponentIds)
 	{
 		QueryConstraint Constraint;
 		Constraint.ComponentConstraint = ComponentId;
-		SnapshotEntitiesConstraint.OrConstraint.Add(Constraint);
+		OrConstraint.OrConstraint.Add(Constraint);
 	}
 
-	return SnapshotEntitiesConstraint;
+	return OrConstraint;
 }
 
-QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
+QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
 {
-	QueryConstraint AlwaysRelevantConstraint;
-
-	Worker_ComponentId ComponentIds[] = { SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID };
+	return CreateOrConstraint({ SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID,
+								SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID, SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID });
+}
 
-	for (Worker_ComponentId ComponentId : ComponentIds)
-	{
-		QueryConstraint Constraint;
-		Constraint.ComponentConstraint = ComponentId;
-		AlwaysRelevantConstraint.OrConstraint.Add(Constraint);
-	}
+QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
+{
+	return CreateOrConstraint({ SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID });
+}
 
-	return AlwaysRelevantConstraint;
+QueryConstraint InterestFactory::CreateServerOnlyAlwaysRelevantConstraint() const
```

Comment:
```suggestion
QueryConstraint InterestFactory::CreateServerAlwaysRelevantConstraint() const
```
At first I thought this function would ONLY do the ServerOnly component. However, not sure if my suggestion is good.

Reply:
(My thinking is that this is the Server's interest constraint, rather than the constraint including the ServerOnlyAlwaysRelevantComponent)

Reply:
Ah right - yeah this is the Server's AlwaysRelevant constraint, which includes both components.
I'll try to come up with a better name.

Reply:
Renamed the functions to CreateServerAlwaysRelevant..., and CreateClientAlwaysRelevant...